src/main/java/hudson/remoting/Channel.java

```diff
@@ -1499,6 +1594,18 @@ public static Channel current() {
     }
 
     /**
+     * Calls {@link #dumpDiagnostics(PrintWriter)} across all the active channels in this system.
+     * Used for diagnostics.
+     */
+    public static void dumpDiagnosticsForAll(PrintWriter w) throws IOException {
+        for (Ref ref : ACTIVE_CHANNELS.values()) {
```

Comment:
I think it would be better to use `ACTIVE_CHANNELS.values().toArray(new Ref[0])` so that we do not need the lock. You risk CME iterating, e.g. see the javadoc for `Collections.synchronizedMap` which explicitly states:

> It is imperative that the user manually synchronize on the returned map when iterating over any of its collection views

The `toArray` will give you the shortest lock time